bitnami/mongodb/templates/standalone/scripts-configmap.yaml

```diff
@@ -0,0 +1,26 @@
+{{- if eq .Values.architecture "standalone" }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "mongodb.fullname" . }}-scripts
+  namespace: {{ include "mongodb.namespace" . }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/component: mongodb
+data:
+  {{- $fullname := include "mongodb.fullname" . }}
+  {{- $releaseNamespace := include "mongodb.namespace" . }} 
+  setup.sh: |-
+    #!/bin/bash
+    if [[ -n "$VAULT_CREDENTIAL_FILE" ]] && [[ -f "$VAULT_CREDENTIAL_FILE" ]]; then
+        while read -r line; do
+        line=$(echo "$line" | sed 's/: /=/g')
+        declare $line; 
+        done < "$VAULT_CREDENTIAL_FILE"
+        export MONGODB_ROOT_PASSWORD="$MONGODB_ROOT_PASSWORD"
+        export MONGODB_USERNAME="$MONGODB_USERNAME"
+        export MONGODB_PASSWORD="$MONGODB_PASSWORD"
+        export MONGODB_DATABASE="$MONGODB_DATABASE"
+        export MONGODB_USER_ROLE="${MONGODB_USER_ROLE:-readWrite}"
+    fi
+    exec /opt/bitnami/scripts/mongodb/entrypoint.sh /opt/bitnami/scripts/mongodb/run.sh
+{{- end }}
```

Comment:
This logic should be managed by the container. I mean, if we need to support a way to retrieve the credentials from a file (instead of the env. variables `MONGODB_ROOT_PASSWORD`, `MONGODB_USERNAME`, `MONGODB_PASSWORD`, etc.) that logic should be implemented at container level.

- https://github.com/bitnami/bitnami-docker-mongodb

Reply:
There was a challenge implementing it container level. The details are not on top of my mind, will find the details and share. Will revisit and try to find a way if we can address those.